lib/mixlib/install/backend/artifactory.rb

```diff
@@ -54,24 +56,31 @@ def info
           end
         end
 
-        private
+        def product_versions
+          params = {
+            g: "com.getchef",
+            a: options.product_name,
+            repos: ARTIFACTORY_REPOSITORY
+          }
+          results = client.get("/api/search/versions", params)["results"]
 
-        def artifact(result)
-          ArtifactInfo.new(
-            md5:              result["properties"]["omnibus.md5"].first,
-            sha256:           result["properties"]["omnibus.sha256"].first,
-            version:          result["properties"]["omnibus.version"].first,
-            platform:         result["properties"]["omnibus.platform"].first,
-            platform_version: result["properties"]["omnibus.platform_version"].first,
-            architecture:     result["properties"]["omnibus.architecture"].first,
-            url:              result["downloadUri"]
-          )
+          results.collect { |result| result["version"] }
+        end
+
+        def product_version
+          if options.latest_version?
+            Mixlib::Versioning.find_target_version(product_versions, nil, false, true).to_s
```

Comment:
This is sweet @patrick-wright!

In omnitruck we determine the version like this https://github.com/chef/omnitruck/blob/master/lib/chef/version_resolver.rb#L170-L174 . We might want to look into mixlib-versioning and see if there is anything different. Maybe we can provide the same support as omnitruck does. 

Awesome job!